File: frontend/src/types/TipoGasto.ts
export interface KeywordPair {
    centro_costo: string | null
    concepto: string | null
}

export interface TipoGasto {
    id: number
    tipo: string
    descripcion?: string | null
    indicador_default: string | null
    excluir_presupuesto: boolean
    activo: boolean
    keywords: KeywordPair[]
    prioridad: number
    direccion: 'ingreso' | 'egreso'
}

File: Backend/src/domain/models/tipo_gasto.py
from dataclasses import dataclass, field
from typing import Optional, List, Dict, Any
from datetime import datetime


@dataclass
class TipoGasto:
    """Entidad de Dominio para tipos de gasto configurables"""
    tipo: str
    descripcion: Optional[str] = None
    indicador_default: Optional[str] = None
    excluir_presupuesto: bool = False
    activo: bool = True
    keywords: List[Dict[str, Any]] = field(default_factory=list)
    prioridad: int = 99
    id: Optional[int] = None
    created_at: Optional[datetime] = None

    def __post_init__(self):
        if not self.tipo:
            raise ValueError("El tipo de gasto es obligatorio")
